x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -388,6 +382,40 @@ private boolean isCacheEnabled() {
         return jwtCache != null && jwtCacheHelper != null;
     }
 
+    /**
+     * Format and filter JWT contents as user metadata.
+     * @param claimsSet Claims are supported. Claim keys are prefixed by "jwt_claim_".
+     * @return Map of formatted and filtered values to be used as user metadata.
+     */
+    private Map<String, Object> buildUserMetadata(JWTClaimsSet claimsSet) {
+        final HashMap<String, Object> metadata = new HashMap<>(Map.of("jwt_token_type", jwtAuthenticator.getTokenType().value()));
+        if (populateUserMetadata) {
```

Comment:
Always populate the token type metadata because `populateUserMetadata` controls whether metadata should be populated from the "claimSet".